api/docs/multi_trace_window.dox

```diff
@@ -0,0 +1,396 @@
+ *
+ * * Neither the name of Google, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL VMWARE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/**
+ ****************************************************************************
+\page page_multi_trace_window Multi-Window Memtraces
+
+\tableofcontents
+
+# Overview
+
+Memory address traces (or "memtraces") are critical tools for analyzing application
+behavior and performance.  DynamoRIO's \ref page_drcachesim provides memtrace
+gathering and analyzing tools.  Today, these memtraces are gathered either during a
+full execution of a standalone application, or during a short burst of execution on a
+server.  For long-running benchmarks with phased behavior, however, we would like to
+instead gather a series of smaller memtrace samples to ease simulation while still
+representing the application in the aggregate.  This sampling may also be useful for
+servers with phased behavior.
+
+# Initial Use Case: SPEC2017
+
+An initial use case driving this work is the gathering of memtraces from SPEC2017.
+These benchmarks execute tens of trillions of instructions each and include phased
+behavior.  Our goal is to gather uniformly sampled memtrace sequences across the
+whole execution.  To support a full warmup for each sample, our target is 10 billion
+instructions in each.  Something like 50 samples per benchmark should suffice to
+capture major phases.  With these parameters, for a 25-trillion-instruction
+benchmark, that would look like a series of 10 billion instruction traces each
+followed by 490 billion instructions of untraced execution.
+
+# Design Point: Separate Traces v. Merged-with-Markers
+
+Focusing on a use case of a series of 50 10-billion-instruction traces for a SPEC
+benchmark, there are two main ways to store them.  We could create 50 independent
+sets of trace files, each with its own metadata and separate set of sharded data
+files.  A simulator could either simulate all 50 separately and aggregate just the
+resulting statistics, or a single instance of a simulator could fast-forward between
```

Comment:
Right this would only really be done with execution-driven simulation with different levels of detail -- for traces the simulator would just jump ahead and use part of the next sample to warm up caches, etc.  Re-worded.